subprojects/core/src/test/groovy/org/gradle/api/internal/plugins/DefaultPluginContainerTest.groovy

```diff
@@ -103,6 +103,27 @@ class DefaultPluginContainerTest extends Specification {
         !container.hasPlugin(plugin2Class)
     }
 
+    def "offers plugin management via injectable plugin type "() {
+        when:
+        def basePluginClass = classLoader.parseClass("""
+        import org.gradle.api.Plugin
+        import org.gradle.api.Project
+        class TestPlugin1 implements Plugin<Project> {
+          void apply(Project project) {}
+          @javax.inject.Inject
+          public org.gradle.internal.reflect.Instantiator getter() {
+            throw new NullPointerException();
+          }
+        }
+    """)
+        def p = container.apply(basePluginClass)
```

Comment:
Please don't use single-character variable names. This should be called `plugin` or `appliedPlugin`.